### PR TITLE
fix(dashboard): date dropdown initialize

### PR DESCRIPTION
**Pull request by minseolee**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

1. 'Current' month case 
스크린샷 2023-01-13 오후 6 45 55

2. Some month case
스크린샷 2023-01-13 오후 6 46 15

3. 'Custom' month case
스크린샷 2023-01-13 오후 6 48 24

### CODE REVIEW

**Review by piggggggggy — APPROVED** [review]
LGTM!!

File: src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue [hunk 1]
```diff
@@ -128,6 +134,51 @@ export default defineComponent({
...
+            && start.m === end.m
+            && start.d === '01'
+            && (Number(end.d) === dayjs(`${end.y}-${end.m}`).daysInMonth())
+            ) {
+                return state.monthMenuItems.findIndex((d) => d.name === `${start.y}-${start.m}`);
+            }
+
+            // 3. custom => else cases.
+            // Index 14 is 'Custom' menu index.
+            return 14;
```
Comment (yuda110) [inline src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue:176]: suggestion: maybe this?

```suggestion
            return state.monthMenuItems.length - 1;
```

File: src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue [hunk 2]
```diff
@@ -128,6 +134,51 @@ export default defineComponent({
...
+            };
+
+            // 1. default month => start is (month 'current' + day '1'), end is (month 'current + day 'today')
+            // Index 0 is 'Current' menu index
+            if (!start.d || !end.d
+            || (_start === `${current.y}-${current.m}-01`
+            && _end === `${current.y}-${current.m}-${current.d}`)
+            ) {
+                return 0;
+            }
```
Comment (yuda110) [inline src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue:163]: suggestion: I think this is safer
```typescript
const _current = dayjs.utc();
const _start = dayjs.utc(props.dateRange?.start);
const _end = dayjs.utc(props.dateRange?.end);
if (!props.dateRange?.start || !props.dateRange?.end || (_start.isSame(_current.startOf('month'), 'day') && _end.isSame(_current.endOf('month'), 'day'))) { return 0; }
```

File: src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue [hunk 3]
```diff
@@ -128,6 +134,51 @@ export default defineComponent({
...
+
+            // 2. some month => start is (month 'n' + day '1'), end is (month 'n' + day '{last day}')
+            if (start.y === end.y
+            && start.m === end.m
+            && start.d === '01'
+            && (Number(end.d) === dayjs(`${end.y}-${end.m}`).daysInMonth())
+            ) {
+                return state.monthMenuItems.findIndex((d) => d.name === `${start.y}-${start.m}`);
+            }
```
Comment (yuda110) [inline src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue:172]: suggestion:

```suggestion
            if (_start.isSame(_end, 'month') && _start.isSame(_start.startOf('month'), 'day') && _end.isSame(_end.endOf('month'))
            ) {
                return state.monthMenuItems.findIndex((d) => d.name === `${start.y}-${start.m}`);
            }
```

**Review by yuda110 — APPROVED** [review]
LGTM!!! I left some comments!!!!

File: src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue (hunk 2 above)
Comment (yuda110) [reply src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue:163]: `isSame` 
Passing in `month` will check `month` and `year`. Passing in `day` will check `day`, `month`, and `year`.

**Pull request merged**